webapi/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -634,14 +634,18 @@ private async Task<ChatMessage> SaveNewResponseAsync(string response, string pro
     /// Updates previously saved response in the chat history.
     /// </summary>
     /// <param name="updatedResponse">Updated response from the chat.</param>
-    /// <param name="messageId">The chat message ID</param>
+    /// <param name="messageId">The chat message ID.</param>
+    /// <param name="chatId">The chat ID that's used as the partition Id.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
-    private async Task UpdateChatMessageContentAsync(string updatedResponse, string messageId, CancellationToken cancellationToken)
+    private async Task UpdateChatMessageContentAsync(string updatedResponse, string messageId, string chatId, CancellationToken cancellationToken)
     {
-        // Make sure the chat exists.
-        var chatMessage = await this._chatMessageRepository.FindByIdAsync(messageId);
-        chatMessage.Content = updatedResponse;
+        ChatMessage? chatMessage = null;
+        if (!await this._chatMessageRepository.TryFindByIdAsync(messageId, chatId, callback: v => chatMessage = v))
+        {
+            throw new ArgumentException($"Chat message {messageId} does not exist.");
+        }
 
+        chatMessage!.Content = updatedResponse;
         await this._chatMessageRepository.UpsertAsync(chatMessage);
```

Comment:
Aux: This upsert isn't including explicit partition key

Reply:
Since we're updating an existing entity, the partitionKey should already exist on the ChatMessage object and extracted in CosmosDbContext

Reply:
FWIW - Upsert works the same way wrt partition key extraction for create or update (because the entire entity is present)

Reply:
Totally, that's why I added the partition key to the call explicitly in CosmosDBContext. I'll add it to the Create call there for consistency too